### PR TITLE
Refactor parsing completions

### PR DESCRIPTION
**Pull request by antaz**
Refactor parsing methods into a mixin and memoize the result for the completion object

### CODE REVIEW

File: lib/llm/response/completion.rb [hunk 1]
```diff
@@ -6,37 +6,47 @@ class Response::Completion < Response
...
       @provider.respond_to?(:completion_total_tokens) ?
-        @provider.__send__(:completion_total_tokens, raw) :
+        parsed[:total_tokens] :
         prompt_tokens + completion_tokens
     end
+
+    private
+
+    ##
+    # @private
```
Comment (0x1eef) [inline lib/llm/response/completion.rb:45]: No need to add `@private` here. Yardoc will use Ruby's visibility rules, and this method is already private.

File: lib/llm/providers/anthropic.rb [hunk 2]
```diff
@@ -5,6 +5,22 @@ module LLM
   # The Anthropic class implements a provider for
   # [Anthropic](https://www.anthropic.com)
   class Anthropic < Provider
+    module Parsable
+      def parse_completion(raw)
```
Comment (0x1eef) [inline lib/llm/providers/anthropic.rb:9]: We're missing documentation for this method.

Comment (0x1eef) [inline lib/llm/providers/anthropic.rb:9]: A completion would be an instance of `LLM::Completion`, and `raw` is just a tiny fragment of it. So I'm not sure a method with the name `parse_completion` and argument of `raw` is the best choice of words.

**Review by 0x1eef — APPROVED** [review]
Nice work :+1: I left some comments / thoughts

File: lib/llm/providers/anthropic.rb [hunk 3]
```diff
@@ -5,6 +5,22 @@ module LLM
   # The Anthropic class implements a provider for
   # [Anthropic](https://www.anthropic.com)
   class Anthropic < Provider
+    module Parsable
```
[the remark this answers is not in the document]
Comment (antaz) [reply lib/llm/providers/anthropic.rb:8]: @0x1eef refer to #24

File: lib/llm/providers/anthropic.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/providers/anthropic.rb:9]: @0x1eef I agree on the bad naming. do you have any suggestions? regarding the naming and organization of Parseable and its methods

**Pull request merged**